syntax = "proto3";
package monitor.proto;

import "google/protobuf/empty.proto";
import "net_info.proto";
import "mem_info.proto";
import "cpu_stat.proto";
import "cpu_softirq.proto";
import "cpu_load.proto";
import "disk_info.proto";

// message MemInfo {
//   int64 total = 1;
//   int64 free = 2;
//   int64 avail= 3;
//   int64 buffers = 4;
//   int64 cached = 5;
//   int64 swap_cached = 6;
//   int64 active = 7;
//   int64 inactive = 8;
//   int64 active_anon = 9;
//   int64 inactive_anon = 10;
//   int64 active_file = 11;
//   int64 inactive_file = 12;
//   int64 dirty = 13;
//   int64 writeback = 14;
//   int64 anon_pages = 15;
//   int64 mapped = 16;
//   int64 kReclaimable = 17;
//   int64 sReclaimable = 18;
//   int64 sUnreclaim = 19;
//   int32 used_percent = 31;
// }

// message CpuStat{
//   string cpu_name = 1;
//   float cpu_percent = 2;
//   float usr_percent = 3;
//   float system_percent = 4;
//   float nice_percent = 5;
//   float idle_percent = 6;
//   float io_wait_percent = 7;
//   float irq_percent = 8;
//   float soft_irq_percent = 9;
// }
// message SoftIrq {
//   string cpu = 1;
//   float hi = 2;
//   float timer = 3;
//   float net_tx = 4;
//   float net_rx = 5;
//   float block = 6;
//   float irq_poll = 7;
//   float tasklet = 8;
//   float sched = 9;
//   float hrtimer = 10;
//   float rcu = 11;
// }

// message CpuLoad {
//   float load_avg_1 = 1;
//   float load_avg_3 = 2;
//   float load_avg_15 = 3;
// }



// message NetInfo {
//   string name = 1;
//   int64 send_rate = 2;
//   int64 rcv_rate = 3;
//   int64 send_packets_rate = 4;
//   int64 rcv_packets_rate = 5;
// }





// 主机标识信息 - 用于调度服务器识别和定位服务器
message HostInfo {
  string hostname = 1;       // 主机名
  string ip_address = 2;     // IP 地址（主网卡 IP）
}

message MonitorInfo{
  string name = 1;           // 主机名（保留兼容）
  HostInfo host_info = 2;    // 主机标识信息
  repeated SoftIrq soft_irq = 4;
  CpuLoad cpu_load = 5;
  repeated CpuStat cpu_stat = 6;
  MemInfo mem_info = 7;
  repeated NetInfo net_info = 8;
  repeated DiskInfo disk_info = 9;
}

service GrpcManager {
  // 工作者推送监控数据给管理者
  rpc SetMonitorInfo(MonitorInfo) returns (google.protobuf.Empty) {
  }
  
  // 保留：管理者主动获取监控数据（可选）
  rpc GetMonitorInfo(google.protobuf.Empty) returns (MonitorInfo) {
  }
}



